vllm/tracing.py

```diff
@@ -0,0 +1,31 @@
+import opentelemetry.semconv.ai
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import (
+    OTLPSpanExporter)
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from opentelemetry.trace import set_tracer_provider
+
+
+def init_tracer(instrumenting_module_name):
+    trace_provider = TracerProvider()
+
+    # The endpoint of OTLPSpanExporter is set from envvars:
+    #  OTEL_EXPORTER_OTLP_ENDPOINT
+    #  OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
+    trace_provider.add_span_processor(BatchSpanProcessor(OTLPSpanExporter()))
+    set_tracer_provider(trace_provider)
+
+    tracer = trace_provider.get_tracer(instrumenting_module_name)
+    return tracer
```

Comment:
Instead the SpanProcesser should be setup as part of the initialization process of the OpenAI-compatible server.

Reply:
Thanks. Good catch. Has introducing the `--otlp-endpoint` flag solved this issue?

Reply:
> Has introducing the --otlp-endpoint flag solved this issue?

Not entirely :). The SpanProcesser is still set up in `init_tracer` via `LLMEngine.__init__`, which still results in a SpanProcessor being added when using VLLM as a library.

The following chunk:

```python
    trace_provider = TracerProvider()

    # The endpoint of OTLPSpanExporter is set from envvars:
    #  OTEL_EXPORTER_OTLP_ENDPOINT
    #  OTEL_EXPORTER_OTLP_TRACES_ENDPOINT
    trace_provider.add_span_processor(BatchSpanProcessor(OTLPSpanExporter()))
    set_tracer_provider(trace_provider)
```
should be moved to the initialization specific to the OpenAI-compatible server (e.g. [here](https://github.com/ronensc/vllm/blob/7f5cd5a7ac4f760bfc87f56f01153d691edb7496/vllm/entrypoints/api_server.py#L109)).

Reply:
`init_tracer` is only called when `--otlp-endpoint` is set, of course. But we are using VLLM as a library and would also like to use the OpenTelemetry instrumentation introduced here :).

Reply:
To configure the `otlp_endpoint` flag for offline inferencing, use the following syntax:
```python
from vllm import LLM, SamplingParams

llm = LLM(
	model="facebook/opt-125m", 
	disable_log_stats=False, 
	otlp_endpoint="http://172.17.0.2:4318/v1/traces",
)
```
However, it's important to note that currently, no traces will be reported because `trace_context` is set to None:
https://github.com/vllm-project/vllm/pull/4687/files#diff-c89ac25bd066e936e80260d21be63c7d2379cfedc371a9ff288fb5ba02ae1350R1003-R1004

I'll work on implementing a fix for it.